### PR TITLE
SDA-4457 : add addon inquiries endpoints get and list

### PR DESCRIPTION
**Pull request by ciaranRoche**
Part of : https://issues.redhat.com/browse/SDA-4457

### CODE REVIEW

**Comment by ciaranRoche** [comment]
@tzvatot @vkareh PTAL

File: model/clusters_mgmt/v1/add_on_inqueries_resource.model [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        // add-ons with a name starting with `my` the value should be:
+        //
+        // [source,sql]
+        // ----
+        // name like 'my%'
+        // ----
+        //
+        // If the parameter isn't provided, or if the value is empty, then all the add-ons
+        // that the user has permission to see will be returned.
+        in Search String
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/add_on_inqueries_resource.model:41]: Do we actually support it in the backend for addon inqueries? or are you just adding it for compatibility reasons?

Comment (ciaranRoche) [reply model/clusters_mgmt/v1/add_on_inqueries_resource.model:41]: Its the same as a normal list addons. So has the same list args. We fetch the cluster, list the addons, then do the inquery on the resulting list of addons on the cluster, and return the result.

File: model/clusters_mgmt/v1/add_on_inqueries_resource.model [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+        // regardless of the size of the page.
+        out Total Integer
+
+        // Retrieved list of add-ons.
+        out Items []AddOn
+    }
+
+    locator AddonInquery {
+        target AddonInquery
+        variable AddOnID
```
Comment (tzvatot) [inline model/clusters_mgmt/v1/add_on_inqueries_resource.model:69]: Can we make it consistent lowercase 'AddonID' or is there some limitation?

Comment (ciaranRoche) [reply model/clusters_mgmt/v1/add_on_inqueries_resource.model]: done :+1:

**Comment by vkareh** [comment]
LGTM

**Pull request merged**